src/overload.ts

```diff
@@ -1,130 +1,365 @@
+    request: T,
+    client: ClientType,
 ): T {
-    const originalLog = client.log.bind(client);
-
-    const _overloadedLog = async (
-        request: T extends Flows
-            ? FlowLogRequest
-            : T extends Prompts
-              ? PromptLogRequest
-              : T extends Tools
-                ? ToolLogRequest
-                : T extends Evaluators
-                  ? CreateEvaluatorLogRequest
-                  : never,
-        options?: T extends Flows
-            ? Flows.RequestOptions
-            : T extends Prompts
-              ? Prompts.RequestOptions
-              : T extends Tools
-                ? Tools.RequestOptions
-                : T extends Evaluators
-                  ? Evaluators.RequestOptions
-                  : never,
-    ) => {
-        const traceId = getTraceId();
-        if (traceId !== undefined && client instanceof Flows) {
+    const traceId = getTraceId();
+    if (traceId !== undefined) {
+        if (client instanceof Flows) {
             const context = getDecoratorContext();
             if (context === undefined) {
                 throw new HumanloopRuntimeError(
                     "Internal error: trace_id context is set outside a decorator context.",
                 );
             }
             throw new HumanloopRuntimeError(
-                `Using flows.log() is not allowed: Flow decorator for File ${context.path} manages the tracing and trace completion.`,
+                `Using \`flows.log()\` is not allowed: Flow decorator ` +
+                `for File ${context.path} manages the tracing and trace completion.`,
             );
         }
 
-        if (traceId !== undefined) {
-            if ("traceParentId" in request) {
-                console.warn(
-                    "Ignoring trace_parent_id argument: the Flow decorator manages tracing.",
+        if ("traceParentId" in request) {
+            console.warn(
+                "Ignoring trace_parent_id argument: the Flow decorator manages tracing.",
+            );
+        }
+        return {
+            ...request,
+            traceParentId: traceId,
+        };
+    }
+    return request;
+}
+
+/**
+ * Load prompt/agent file content from local filesystem into API request.
+ *
+ * Retrieves the file content at the specified path and adds it to request
+ * under the appropriate field ('prompt' or 'agent'), allowing local files
+ * to be used in API calls instead of fetching from Humanloop API.
+ *
+ * @param request API request object
+ * @param client Client instance making the call
+ * @param fileSyncer FileSyncer handling local file operations
+ * @returns Updated request with file content in the appropriate field
+ * @throws HumanloopRuntimeError On validation or file loading failures.
+ *   For example, an invalid path format (absolute paths, leading/trailing slashes, etc.) 
+ *   or a file not being found.
+ */
+function handleLocalFiles<T extends LogRequestType>(
```

Comment:
There's some similar logic in resolve File in eval.run(...). Modifying it now could take a while, but do add it in the Linear project for milestone 2